core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -518,6 +518,9 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     _taskScheduler.postStartHook()
     _env.metricsSystem.registerSource(new DAGSchedulerSource(dagScheduler))
     _env.metricsSystem.registerSource(new BlockManagerSource(_env.blockManager))
+    _executorAllocationManager.foreach { e =>
```

Comment:
Hi @sryza , thanks a lot for your comments. The reason I put this metric source into `ExecutorAllocationManager` is that I don't want to change `ExecutorAllocationManager` a lot to expose the internal status to metric source, so I just wrote this metric source as the inner class. Do you have specific concern about this implementation?